### ISSUE TITLE
Deadkey combo " ' + C " will result in "Ć" instead of "Ç" in RStudio. *All* other programs return "Ç" right

### ISSUE THREAD

**Issue by dvictori** [rstudio/rstudio#10125]
### System details

    RStudio Edition : Desktop
    RStudio Version : RStudio 2021.09.1+372 "Ghost Orchid" Release (8b9ced188245155642d024aa3630363df611088a, 2021-11-08) for Ubuntu Bionic Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) QtWebEngine/5.12.8 Chrome/69.0.3497.128 Safari/537.36  
    OS Version      : Ubuntu 20.04.3 LTS
    R Version       : 4.1.2 (2021-11-01)

### Steps to reproduce the problem

Using Ubuntu with deadkeys keyboard setup. Anywhere on my system, **'+c** (accent + c) will give me **ç** (cedilla). O RStudio desktop, I get **ć**.

### Describe the problem in detail

Mlab reported this problem in the Rstudio community and was instructed to open an issue here. But I could not find the issue. So I'm opening one.

The post from Rstudio community is:
_Deadkey combo " ' + C " will result in "Ć" instead of "Ç" in RStudio. *All* other programs return "Ç" right_
https://community.rstudio.com/t/deadkey-combo-c-will-result-in-c-instead-of-c-in-rstudio-all-other-programs-return-c-right/58225

### Describe the behavior you expected

I should get **ç** instead of **ć**.

- [x] I have read the guide for [submitting good bug reports](https://github.com/rstudio/rstudio/wiki/Writing-Good-Bug-Reports).
- [X] I have installed the latest version of RStudio, and confirmed that the issue still persists.
- [ ] If I am reporting an RStudio crash, I have included a [diagnostics report](https://support.rstudio.com/hc/en-us/articles/200321257-Running-a-Diagnostics-Report).
- [ ] I have done my best to include a minimal, self-contained set of instructions for consistently reproducing the issue.

**Comment by ronblum** [comment]
@dvictori Thank you for raising this! Interesting: I'm using Ubuntu 20.04.3 with English-US-Intl-Deadkeys. But when I run with other applications such as Terminal and LibreOffice Writer, I get _ć_ when I use <kbd>'+c</kbd>. To get a _ç_, I need to use <kbd>ALT+,</kbd>. This probably doesn't make a difference but what desktop environment are you using?

**Comment by dvictori** [comment]
@ronblum I'm using the default desktop (Gnome). But i tweaked things a bit and just found out what happened and how to fix this.

TL;DR:  
adding `export LC_CTYPE=pt_BR.UTF-8 to `~/.profile` fixed things.

Long story:  
For me `alt+,` will also give _ç_ but I'm used to typing it as `'+c`. So in order to change this behavior I saw some people commenting that one should place the line `GTK_IM_MODULE=cedilla` in `/etc/environment`. This changed the behavior of the `'+c` combo everywhere except on RStudio (afaik).

But I just found out that Ubuntu comes with `/etc/profile.d/cedilla-portuguese.sh` which changes another env variable when the system formats are set to _Brasil_ or _Portugal_. There the variable to change is `LC_CTYPE`. Since I'm using US format, this variable was not affected. Adding `export LC_CTYPE=pt_BR.UTF-8 to `~/.profile` fixed things. And so far, no side effects.

**Comment by ronblum** [comment]
@dvictori Thank you very much for doing the research and providing the solution!

**Comment by lisboarocha** [comment]
@dvictori 

I have the same problem and I've tried your solution, but it does not change anything (neither with dead keys, neither with alt. intl). The problem also appears on the login screen, where password does not accept `'+c`

At ~/.profile, i've put the line in the `if's` and out of it, but anything have worked.

do you have any idea?

thanks

**Comment by lisboarocha** [comment]
Thank you for this, @dvictori.
I'm also using Ubuntu 20.04, but anything works here.
I've even tried to directly change the /usr/share/X11/locale/en_US.UTF-8/compose file, but it did also not work.
I'm so going to use `alt+,`.
Thank you very much anyway!
